app/models/secret_storage.rb

```diff
@@ -50,6 +50,51 @@ def self.keys
     end
   end
 
+  class HashicorpVault
+
+    VAULT_SECRET_BACKEND ='secret/'.freeze
+    # we don't really want other directories in here,
+    # and there may be other chars that we find we don't like
+    ENCODINGS = {"/": "%2F"}
+
+    def self.read(key)
+      result = Vault.logical.read(vault_path(key))
+      return false if result.nil?
+      result = result.dup.to_h
+      # making sure that the delete won't blow up
+      result[:value] = result[:data][:vault]
```

Comment:
that now has a hash with another nested hash and the secret is in both ...

result.merge(result.delete(:data))
result[:value] = result.delete(:vault)